src/lib/react-server-route.js

```diff
@@ -58,24 +64,42 @@ module.exports = function(req, res, next) {
   // lead to render properties that can be used to generate page components.
   match({ routes, location }, (error, redirectLocation, renderProps) => {
     // obviously, we need an error handler.
-    if (error) { res.status(500).send(error.message); }
+    if (error) {
+      res.status(500).send(error.message);
+      return;
+    }
 
     // React router lets you specify redirects. If we had any, we literally
     // just tell our server that we need to look up a different URL.
-    else if (redirectLocation) {
+    if (redirectLocation) {
       res.redirect(302, redirectLocation.pathname + redirectLocation.search);
     }
-
     // This is the most interesting part: we have content that React can render.
     else if (renderProps) {
+
+      locale = getLocale(req.headers["accept-language"], firstPath);
+      if (location === "/") {
+        res.redirect(302, location + locale + search);
+        return;
+      }
+      function createElement(Component, props) {
+        var messages = locales[locale];
+        // make sure you pass all the props in!
+        return (
+          <IntlProvider locale={locale} messages={messages}>
```

Comment:
This is similar to what we do client side. We can do server side rendering for the strings this way, which, after cloudfront cache, and react rendering, it's pretty fast to build, send, and render this.